src/transformers/pipelines/base.py

```diff
@@ -83,24 +92,98 @@ def infer_framework_from_model(
         )
     if isinstance(model, str):
         model_kwargs["_from_pipeline"] = task
-        if is_torch_available() and not is_tf_available():
-            model_class = model_classes.get("pt", AutoModel)
-            model = model_class.from_pretrained(model, **model_kwargs)
-        elif is_tf_available() and not is_torch_available():
-            model_class = model_classes.get("tf", TFAutoModel)
-            model = model_class.from_pretrained(model, **model_kwargs)
-        else:
+        class_tuple = ()
+        look_pt = is_torch_available() and framework in {"pt", None}
+        look_tf = is_tf_available() and framework in {"tf", None}
+        if model_classes:
+            if look_pt:
+                class_tuple = class_tuple + model_classes.get("pt", (AutoModel,))
+            if look_tf:
+                class_tuple = class_tuple + model_classes.get("tf", (TFAutoModel,))
+        if config.architectures:
+            klasses = []
```

Comment:
(nit) don't really like the name `klasses` -> is that a common naming?

Reply:
It's quite common when you need `class` as a variable name. 

A bit of background
https://news.ycombinator.com/item?id=19690657c

`cls` is ok for single, but for plural it becomes `clss` which is a bit confusing, `_class(es)` is another proposal (I did not encounter it that much tbh, but it's fine to me)

Reply:
Because naming uniformity beats correctness everywhere in my mind.
`klass` and `klasses` obviously go together. If I used `classes` then the variable has to be named `_class` (or  `class_`).

Reply:
Renamed everything `classes` and `_class` as it seems better for consensus !